backend/stream.go

```diff
@@ -2,30 +2,53 @@ package backend
 
 import (
 	"context"
+	"encoding/json"
 )
 
 // StreamHandler handles streams.
 // This is EXPERIMENTAL and is a subject to change till Grafana 8.
 type StreamHandler interface {
-	// Called when a user tries to connect to a plugin/datasource managed channel.
-	CanSubscribeToStream(context.Context, *SubscribeToStreamRequest) (*SubscribeToStreamResponse, error)
-	// RunStream will be initiated by Grafana to consume a stream from a plugin.
-	// For streams with keepalive set this will only be called once the first client
-	// successfully subscribed to a stream channel. And when there are no longer any
-	// subscribers, the call will be terminated by Grafana.
+	// SubscribeStream called when a user tries to subscribe to a plugin/datasource
+	// managed channel path.
+	SubscribeStream(context.Context, *SubscribeStreamRequest) (*SubscribeStreamResponse, error)
+	// PublishStream called when a user tries to publish to a plugin/datasource
+	// managed channel path.
+	PublishStream(context.Context, *PublishStreamRequest) (*PublishStreamResponse, error)
+	// RunStream will be initiated by Grafana to consume a stream where keepalive
+	// option set to true. In this case RunStream will only be called once for the
+	// first client successfully subscribed to a channel path. When Grafana detects
+	// that there are no longer any subscribers inside a channel, the call will be
+	// terminated until next active subscriber appears. Call termination can happen
+	// with a delay.
 	RunStream(context.Context, *RunStreamRequest, StreamPacketSender) error
 }
 
-// SubscribeToStreamRequest is EXPERIMENTAL and is a subject to change till Grafana 8.
-type SubscribeToStreamRequest struct {
+// SubscribeStreamRequest is EXPERIMENTAL and is a subject to change till Grafana 8.
+type SubscribeStreamRequest struct {
 	PluginContext PluginContext
 	Path          string
 }
 
-// SubscribeToStreamResponse is EXPERIMENTAL and is a subject to change till Grafana 8.
-type SubscribeToStreamResponse struct {
-	OK      bool
-	Message string
+// SubscribeStreamResponse is EXPERIMENTAL and is a subject to change till Grafana 8.
+type SubscribeStreamResponse struct {
+	OK           bool
+	ErrorMessage string
+	Schema       json.RawMessage
```

Comment:
maybe just "Data"?  like everywhere else -- and it is returned in the initial request